### PR TITLE
Move GitHub Repo logic into Store

### PR DESCRIPTION
**Pull request by keegancsmith**
See commit messages. The most important commit is `	store: Move setRepoFieldsFromRemote to store`

Fixes https://app.asana.com/0/87040567695724/147734985562458
Fixes https://app.asana.com/0/146253180133821/137848642885286
Fixes probably a bunch of other issues

### CODE REVIEW

File: services/backend/internal/localstore/repos.go [hunk 1]
```diff
@@ -661,3 +658,108 @@ func (s *repos) Delete(ctx context.Context, repo int32) error {
...
+
+	// repo.ID != 0 since we may be called from Create
+	if updateWithDataFromRemote != nil && repo.ID != 0 {
+		log15.Debug("Updating repo metadata from remote", "repo", repo.URI)
+		// setRepoFieldsFromRemote is used in read requests, including
+		// unauthed ones. However, this write isn't as the user, but
+		// rather an optimization for us to save the data from
+		// github. As such we use an elevated context to allow the
+		// write.
+		if err := store.ReposFromContext(ctx).Update(elevatedActor(ctx), *updateWithDataFromRemote); err != nil {
```
Comment (neelance) [inline services/backend/internal/localstore/repos.go]: This code is already being executed inside of the repos store. Can't it just call `update` locally without the need for an elevated actor?

File: services/backend/internal/localstore/repos.go [hunk 2]
```diff
@@ -171,24 +172,29 @@ func (s *repos) Get(ctx context.Context, id int32) (*sourcegraph.Repo, error) {
...
 }
 
 // finishGetRepo checks permissions and fills in additional fields on
 // repo. It MUST be called after fetching a repo from the DB before
 // returning the repo.
-func finishGetRepo(ctx context.Context, repo *sourcegraph.Repo) (*sourcegraph.Repo, error) {
+//
+// NOTE: The repo returned is the same as the repo passed in. Provided as a
+// convenience.
+func (s *repos) finishGetRepo(ctx context.Context, repo *sourcegraph.Repo) (*sourcegraph.Repo, error) {
```
Comment (neelance) [inline services/backend/internal/localstore/repos.go]: Not related to this PR: I think it is a security risk that the name `finishGetRepo` does not indicate that this includes a critical access check.

**Comment by neelance** [comment]
Added comments. I would like to see them answered before I give LGTM.

**Comment by keegancsmith** [comment]
This conflicts quite badly with the recently merged in and long-lived `search-flow` branch. A bit frustrating since the conflicts seem to come from commits a week ago that are independent of the purpose of the branch.

I'll salvage some of this branch onto a new PR, but will likely abandon the more complicated and important change until another time.

**Comment by keegancsmith** [comment]
https://github.com/sourcegraph/sourcegraph/pull/203 contains the easy to salvage changes. I'll apply any feedback you left that applies to those commits @neelance.

Otherwise I'll tackle this again next week.

File: services/backend/internal/localstore/repos.go (hunk 1 above)
Comment (keegancsmith) [inline services/backend/internal/localstore/repos.go]: a later commit added calling just `update` and removing the elevatedActor.

**Comment by keegancsmith** [comment]
I think I was too quick to blame `search-flow`. I'll take a closer look.

**Pull request closed without merging**